network/src/transport.rs

```diff
@@ -207,193 +218,948 @@ pub async fn perform_handshake<T: TSocket>(
     }
 }
 
-pub fn build_memory_noise_transport(
-    identity_key: x25519::PrivateKey,
-    trusted_peers: Arc<RwLock<HashMap<PeerId, NetworkPublicKeys>>>,
-    application_protocols: SupportedProtocols,
-) -> boxed::BoxedTransport<Connection<impl TSocket>, impl ::std::error::Error> {
-    let memory_transport = memory::MemoryTransport::default();
-    let noise_config = Arc::new(NoiseWrapper::new(identity_key));
-    let noise_timestamps = Arc::new(RwLock::new(AntiReplayTimestamps::default()));
-    let mut own_handshake = HandshakeMsg::new();
-    own_handshake.add(SUPPORTED_MESSAGING_PROTOCOL, application_protocols);
-
-    memory_transport
-        .and_then(move |socket, addr, origin| async move {
-            let remote_public_key = addr.find_noise_proto();
-            let (remote_static_key, socket) = noise_config
-                .upgrade_connection(
-                    socket,
-                    origin,
-                    Some(noise_timestamps),
-                    remote_public_key,
-                    Some(&trusted_peers),
-                )
-                .await?;
-
-            if let Some(peer_id) =
-                identity_key_to_peer_id(&trusted_peers, remote_static_key.as_slice())
-            {
-                Ok((peer_id, socket))
-            } else {
-                Err(io::Error::new(io::ErrorKind::Other, "Not a trusted peer"))
-            }
-        })
-        .and_then(move |(peer_id, socket), addr, origin| async move {
-            perform_handshake(peer_id, socket, addr, origin, &own_handshake).await
-        })
-        .with_timeout(TRANSPORT_TIMEOUT)
-        .boxed()
+/// Check that the parsed `handshake_version` from the `NetworkAddress` is supported.
+/// Return `Err` otherwise.
+fn verify_handshake_version_supported(expected_version: u8, version: u8) -> io::Result<()> {
```

Comment:
it seems weird to have this as its own function when it's just an equality check. Let's inline it :)